.circleci/config.yml

```diff
@@ -1,37 +1,319 @@
+  # Job that runs the unit tests on Browserstack. The browsers that will be used
+  # to run the unit tests on Browserstack are set in: test/browser-providers.js
+  # ----------------------------------------------------------------------------
+  tests_browserstack:
+    <<: *job_defaults
+    resource_class: xlarge
+    environment:
+      BROWSER_STACK_USERNAME: "angularteam1"
+      BROWSER_STACK_ACCESS_KEY: "CaXMeMHD9pr5PHg8N7Jq"
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: ./scripts/circleci/run-browserstack-tests.sh
+
+  # ----------------------------------------------------------------------------
+  # Job that runs the unit tests on Saucelabs. The browsers that will be used
+  # to run the unit tests on Saucelabs are set in: test/browser-providers.js
+  # ----------------------------------------------------------------------------
+  tests_saucelabs:
+    <<: *job_defaults
+    resource_class: xlarge
+    environment:
+      SAUCE_USERNAME: "angular-ci"
+      SAUCE_ACCESS_KEY: "9b988f434ff8-fbca-8aa4-4ae3-35442987"
+      # Note: This number should not be too high because otherwise we might run into
+      # a rate limit exception.
+      KARMA_PARALLEL_BROWSERS: 2
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: ./scripts/circleci/run-saucelabs-tests.sh
 
-      # Specify service dependencies here if necessary
-      # CircleCI maintains a library of pre-built images
-      # documented at https://circleci.com/docs/2.0/circleci-images/
-      # - image: circleci/mongo:3.4.4
+  # --------------------------------------------------
+  # Job that runs the unit tests on the SSR platform
+  # --------------------------------------------------
+  tests_ssr:
+    <<: *job_defaults
+    steps:
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
 
-    working_directory: ~/repo
+      - run: yarn gulp ci:ssr
 
+  # -------------------------------------------------------------------------
+  # Job that pre-render's the universal app with `@angular/platform-server`.
+  # This verifies that Angular Layout can be rendered within Node.
+  # -------------------------------------------------------------------------
+  prerender_build:
+    <<: *job_defaults
     steps:
-      - checkout
+      - *checkout_code
+      - *restore_cache
+      - *yarn_install
+
+      - run: yarn gulp ci:prerender
 
-      # Download and cache dependencies
-      - restore_cache:
-          keys:
-          - v1-dependencies-{{ checksum "yarn.lock" }}
-          # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-
+  # -------------------------------------------------------------------------
+  # Job that makes sure Angular Layout can be integrated with a basic app
+  # -------------------------------------------------------------------------
+  hw_build:
```

Comment:
`hw_build` is pretty unclear at first glance. Please do not abbreviate the job name.

Reply:
Will change to `hello_world_build`